lib/demo.g

```diff
@@ -47,11 +47,11 @@ BindGlobal( "Demonstration", function( file )
     while CHAR_INT( ReadByte( keyboard ) ) <> 'q' do
         storedtime := Runtime();
         result:=READ_COMMAND_REAL( input, true ); # Executing the command.
-        time := Runtime()-storedtime;
+        UpdateStat("time", Runtime()-storedtime);
         if Length(result) = 2 then
-            last3 := last2;
-            last2 := last;
-            last := result[2];
+            UpdateStat("last3", last2);
+            UpdateStat("last2", last);
+            UpdateStat("last", result[2]);
```

Comment:
Since you are touching this code anyway, perhaps you'd also be willing to teach it to update `memory_allocated`, too? Also perfectly fine if you don't want to, of course!

Reply:
I've never actually used this code, and don't even know how it should be used, so I'd prefer to leave it. I've just noticed it has no tests, no documentation, and isn't linked to from anywhere?

Reply:
It is used like this: you take a file and put a bunch of inputs in it, each on its own line. Then you run `Demonstration` on this; whenever you press a key, the next input line is read and executed. The result then looks something like this (where each `demo>` appeared only after a key was pressed):
```
gap> Demonstration("demo.g");

Start of demonstration.

demo> 1+1;
2
demo> Factorial(10);
3628800
demo>
End of demonstration.

gap>
```

Updating last, last2, etc. only is of interest if the demo code references them. So if one wants to write a demo that showcases those, and also `memory_allocated`, then it'd be nice to have `memory_allocated`. But this is nothing this PR needs to do.

I've long wanted to give this code some love, and improve it; e.g. instead of always going to the next command upon any key press, I think only "return" on an empty line should do that; otherwise, it'd be nice if one could show the next command w/o executing it, and also edit it; and other stuff. I think other people implemented such demo modes for other languages, and they can be very useful for classes, presentations during talks etc.